docs/designers-developers/developers/tutorials/javascript/js-build-setup.md

```diff
@@ -0,0 +1,236 @@
+
+### Node Package Manager (npm)
+
+The Node Package Manager (npm) is a tool included with node. npm allows you to install and manage JavaScript packages. npm can also generate and process a special file called `package.json`, which contains some information about your project and the packages your project uses.
+
+To start a new node project, first create a directory to work in.
+
+```
+mkdir myguten-block
+```
+
+You create a new package.json using `npm init`.  This will walk you through creating your package.json file:
+
+```
+$ npm init
+
+package name: (myguten-block) myguten-block
+version: (1.0.0)
+description: Test block
+entry point: (index.js) block.js
+test command:
+git repository:
+keywords:
+author: mkaz
+license: (ISC) GPL-2.0-only
+About to write to /home/mkaz/src/wp/scratch/package.json:
+
+{
+  "name": "myguten-block",
+  "version": "1.0.0",
+  "description": "Test block",
+  "main": "block.js",
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1"
+  },
+  "author": "mkaz",
+  "license": "GPL-2.0-only"
+}
+
+
+Is this OK? (yes) yes
+```
+
+### Using npm to install packages
+
+The next step is to install the packages required. You can install packages using the npm command `npm install`. If you pass the `--save-dev` parameter, npm will write the package as a dev dependency in the package.json file.
+
+Run `npm install --save-dev webpack`
+
+After installing, a `node_modules` directory is created with the webpack module and its dependencies.
+
+Also, if you look at package.json file it will include a new section:
+
+```json
+"dependencies": {
+	"webpack": "^4.29.0"
+}
+```
+
+## Webpack & Babel
+
+You configure webpack to process the `block.js` file and run babel to transform the JSX.
+
+Create the file `webpack.config.js`
+
+```js
+// sets mode webpack runs under
+const NODE_ENV = process.env.NODE_ENV || 'development';
+
+module.exports = {
+	mode: NODE_ENV,
+
+	// entry is the source script
+	entry: './block.js',
+
+	// output is where to write the built file
+	output: {
+		path: __dirname,
+		filename: 'block.build.js',
+	},
+	module: {
+		// the list of rules used to process files
+		// this looks for .js files, exclude files
+		// in node_modules directory, and uses the
+		// babel-loader to process
+		rules: [
+			{
+				test: /.js$/,
+				exclude: /node_modules/,
+				loader: 'babel-loader',
+			},
+		],
+	},
+};
+```
+
+Next, you need to install babel, the webpack loader, and the JSX plugin using:
+
+```
+npm install --save babel-loader babel-core babel-plugin-transform-react-jsx
+```
+
+You configure babel by creating a `.babelrc` file:
+
+```
+{
+	"plugins": [
+		[ "transform-react-jsx", {
+			"pragma": "wp.element.createElement"
+		} ]
+	]
+}
+```
+
+The pragma setting replaces the function JSX uses for transformation, the `wp.element.createElement` is an equivalent wrapper of `React.createElement` which is the default function for JSX.
```

Comment:
I suggest something more concrete, in case the user doesn't have a lot of prior React experience, e.g.

_This pragma setting instructs Babel that any JSX syntax such as `<Hello />` should be transformed into `wp.element.createElement( Hello )`._ The name of the setting (`transform-react-jsx`) is derived from the fact that it overrides the default assumption to transform to `React.createElement( Hello )`.

Reply:
:+1: